### PR TITLE
Add/import publications #1402

### LINKED ISSUE #1492: [templ] Publication import pages

### PR DESCRIPTION
**Pull request by verheyenkoen**
This PR converts the entire add publication wizard (single & multiple) to Templ.
REFS #1402 
FIXES #1492

### CODE REVIEW

File: handlers/publicationcreating/import.go [hunk 1]
```diff
@@ -75,27 +77,29 @@ type YieldValidationErrors struct {
 	Errors form.Errors
 }
 
-func (h *Handler) Add(w http.ResponseWriter, r *http.Request, ctx Context) {
-	tmpl := ""
+func Add(w http.ResponseWriter, r *http.Request, legacyContext Context) {
+	c := ctx.Get(r)
+
+	const pageTitle = "Add - Publications - Biblio"
```
Comment (verheyenkoen) [inline handlers/publicationcreating/import.go:83]: Do we need to pass on the title from the handler or can we do that from the concrete template (add_manual.templ, add_wos.templ, add_bibtex.templ,...)

Comment (nics) [reply handlers/publicationcreating/import.go:83]: @verheyenkoen I just put them in the template

File: handlers/publicationcreating/import.go [hunk 2]
```diff
@@ -151,24 +148,21 @@ func (h *Handler) AddSingleImport(w http.ResponseWriter, r *http.Request, ctx Co
...
-			h.Logger.Warnw("import single publication: could not fetch publication", "errors", err, "publication", b.Identifier, "user", ctx.User.ID)
+			c.Log.Warnw("import single publication: could not fetch publication", "errors", err, "publication", b.Identifier, "user", c.User.ID)
 
 			flash := flash.SimpleFlash().
 				WithLevel("error").
-				WithBody(template.HTML(ctx.Loc.Get("publication.single_import.import_by_id.import_failed")))
+				WithBody(template.HTML(c.Loc.Get("publication.single_import.import_by_id.import_failed")))
 
-			ctx.Flash = append(ctx.Flash, *flash)
+			c.PersistFlash(w, *flash)
```
[the remark this answers is not in the document]
Comment (nicolasfranck) [reply handlers/publicationcreating/import.go:161]: I am experiencing the same problems here on the dataset level

[the remark this answers is not in the document]
Comment (nicolasfranck) [reply handlers/publicationcreating/import.go:161]: looking at the existing examples, most of the code does a redirect to a different url after setting the flash cookie. Might help explain why this is only a problem here. Will investigate this.

[the remark this answers is not in the document]
Comment (nicolasfranck) [reply handlers/publicationcreating/import.go:161]: This works, if you need flash immediately:

```

flash := flash.SimpleFlash().
				WithLevel("error").
				WithTitle("Failed to save draft")
c.Flash = append(c.Flash, *flash)
```

I believe there are other examples for that too.

Seems a bit logical: PersistFlash is to ensure that a next page load picks up errors sent through cookies, and is intended for those situations where the handler cannot send the errors directly in the view.

File: handlers/publicationcreating/import.go [hunk 3]
```diff
@@ -218,34 +206,26 @@ func (h *Handler) AddSingleImport(w http.ResponseWriter, r *http.Request, ctx Co
...
 
-	render.Layout(w, "layouts/default", "publication/pages/add_single_description", YieldAddSingle{
-		Context:        ctx,
-		PageTitle:      "Add - Publications - Biblio",
-		Step:           2,
-		ActiveNav:      "publications",
-		SubNavs:        []string{"description", "files", "contributors", "datasets"},
-		ActiveSubNav:   subNav,
-		Publication:    p,
-		DisplayDetails: displays.PublicationDetails(ctx.User, ctx.Loc, p),
```
[the remark this answers is not in the document]
Comment (nics) [reply handlers/publicationcreating/import.go:229]: yes this is now retrieved with htmx

**Review by nics — APPROVED** [review]

**Pull request merged**